rust/storage/src/lib.rs

```diff
@@ -366,6 +374,19 @@ impl DefaultDispatcher {
         Ok(())
     }
 
+    fn replace_kb(&self, scan_id: &str, kb: Kb) -> Result<(), StorageError> {
+        let mut data = self.kbs.as_ref().write()?;
+        if let Some(scan_entry) = data.get_mut(scan_id) {
+            scan_entry.remove(&kb.key);
+            scan_entry.insert(kb.key.clone(), vec![kb]);
```

Comment:
The call to remove is not necessary, as insert replaces the value automatically and returns the replaced value. I got two solutions:
```suggestion
            if let Some(kb_entry) = scan_entry.get_mut(&kb.key) {
                *kb_entry = vec![kb];
            } else {
                scan_entry.insert(kb.key.clone(), vec![kb]);
            }
```
```suggestion
            scan_entry.insert(kb.key.clone(), vec![kb]);
```
the second one seems a lot cleaner, but is slightly slower, as we do not need the `clone` operation in the first one. I tested a little bit and if the replace function always replaces a value instead of inserting a new one, the first solution is always about 10% faster.